woocommerce/woocommerce-ios
# Blaze: Part 1 of Blaze Ad Destination screen

part of: #11609


## Description


This is the first part of the Blaze Ad destination screen. The main work will be split into several PRs, in order to keep PRs smaller.

This PR in particular has these added:
1. Ad Destination view and viewmodel
2. Navigation from ad preview screen to ad destination screen,
3. A default example parameter added. This is to help demonstrate the parameter view and the estimation calculation.

Not included in this PR and to be tackled in the next set of PRs:
1. Saving and passing data from Ad Destination screen to the Ad preview screen,
2. Parameter functionalities:
    - Actual "Add parameter" functionality (show another screen to enter parameter),
    - swipe to delete parameter,
    - parameter validation,
4. Unit tests.

## Testing instructions


1. Start with a Blaze-eligible site,
6. Go to Products, select a Product,
7. tap "Promote with Blaze",
8. Scroll down to "Ad Destination" menu, tap it,
9. Make sure the "Ad destination" screen is shown.

To test in that screen:
- Cancel button,
- Toggling between the two urls in "Destination URL". Ensure that the checkmark changes correctly. Also ensure that the "Destination" text at the bottom updates with the change.
- Ensure the "Destination" text includes the parameter text.


## Video



https://github.com/woocommerce/woocommerce-ios/assets/266376/4d648b1c-f804-4cc8-81b1-8c248c241d5f


---
